xdsl/pattern_rewriter.py

```diff
@@ -288,7 +304,9 @@ def inline_block_before(
         The block should not be a parent of the operation.
         """
         self.has_done_action = True
-        Rewriter.inline_block_before(block, op, arg_values=arg_values)
+        Rewriter.inline_block_at_location(
+            block, InsertPoint.before(op), arg_values=arg_values
+        )
```

Comment:
I'm not sure what to think here to be honest.
I feel that it was better and more concise previously?

I understand you want to have the API simpler, but in that example I'm not really sure I see why it is beneficial? Having concise methods is really useful for users, as writing everytime `InsertPoint` is a bit annoying in these cases?

I'm wondering what the others are thinking? If they think it's fine I'm fine having this, but I would make that decision consciously at least

Reply:
As a client of APIs, I definitely prefer fewer methods and data structures to think about. Another consideration for me is that `Rewriter` is a lower-level API than `PatternRewriter`, which still has the extended API.

Reply:
Just to make my position clear, I think that the API with a single method and a single datastructure is more ergonomic in general than a collection of helpers.

Reply:
Happy to wait for @AntonLydike and @PapyChacal's opinions, might as well ping @webmiche, @compor, and @zero9178, in case they also have strong opinions.

Reply:
Yeah, we might also just talk about it on Monday for our meeting, as this is probably a general position we should take.
For instance, this would mean removing `rewriter.replace_matched_op` as well, as this is also an API we do not *need*.

Reply:
I'm not sure about that, I think we can also have a clear distinction between a pattern rewriter and a plain rewriter. Also, the matched op seems like a distinct thing, it's no longer about arbitrary locations in code, but about the thing you're locally rewriting.

Reply:
I do like this change; I understand the big discussion about a general direction, I do think it might be overstating it a little, but if I had to choose, I'd rather have the style proposed here by @superlopuh honestly.

Might be a bit overfitted, but when I type with autocompletion, it's cognitively heavy (more than necessary IMO) to read all the variants and pick exactly the right one. With this change, it goes more like "Oh yes, I just want to inline a block", then see I'm asked a location, and pick the option that makes sense for the location *specifically*.

those more specific helpers are just one liners, and can be used in individual rewrites if they save some LOC in specific transformations, IMO :slightly_smiling_face: